plutus-wallet-api/src/Wallet/API.hs

```diff
@@ -475,7 +476,7 @@ createTxAndSubmit range ins outs = do
             , txValidRange = range
             , txSignatures = Map.empty
             }
-    signTxAndSubmit tx
+    signTxAndSubmit $ tx { txFee = minFee tx }
```

Comment:
There's a TODO on this function that should probably go away. Although perhaps there is a new TODO: decide if this is where we should calculate fees or whether it should happen in a different coin selection stage.